Vagrantfile

```diff
@@ -9,14 +9,32 @@ if not File.file?(config_file)
 end
 pubstack_config = YAML::load_file(config_file)
 
-# Validate sites' docroots:
-if pubstack_config.include? 'sites'
+# If the synced folder option is specified, fix the documentroot option
+# for every site.
+if pubstack_config.include? 'synced_folder'
+  # It doesn't make sense to specify both synced_folder and shared_folders,
+  # and the path to the documentroot in the VM will break, so just bail
+  # if both are specified for whatever reason.
+  if pubstack_config.include? 'shared_folders'
+    raise Vagrant::Errors::VagrantError.new, 'You may only use the synced_folder option if shared_folders is not specified. Please fix your config.yml and try again.'
+  end
+
   pubstack_config['sites'].each {|site|
```

Comment:
We don't check for the sites key anywhere else - Any chance you'd be amicable to making that a separate issue?

Re checking docroots - that's correct and also documented in config.example.yml :)